dev/TeachingTip/TeachingTip.cpp

```diff
@@ -1678,21 +1703,55 @@ std::tuple<winrt::Rect, winrt::Thickness, winrt::Thickness> TeachingTip::Determi
     }
 }
 
-std::array<winrt::TeachingTipPlacementMode, 12> TeachingTip::GetPlacementFallbackOrder(winrt::TeachingTipPlacementMode preferredPlacement)
+winrt::Rect TeachingTip::GetEffectiveWindowBounds()
+{
+    if (m_useTestWindowBounds)
+    {
+        return m_testWindowBounds;
+    }
+#ifdef USE_INSIDER_SDK
+    if (winrt::IUIElement10 uiElement10 = *this)
+    {
+        if (auto xamlRoot = uiElement10.XamlRoot())
+        {
+            return winrt::Rect{ 0,0,xamlRoot.Size().Width, xamlRoot.Size().Height };
+        }
+    }
+#endif // USE_INSIDER_SDK
+    return winrt::Window::Current().CoreWindow().Bounds();
+}
+
+winrt::Rect TeachingTip::GetEffectiveScreenBounds()
+{
+    MUX_ASSERT_MSG(!m_returnTopForOutOfWindowPlacement, "When returnTopForOutOfWindowPlacement is true we will never need to get the screen bounds");
+    if (!m_useTestScreenBounds && !ShouldConstrainToRootBounds())
```

Comment:
The way the code is structured, this assert will still fire because we call this API even when we don't need it in DetermineSpaceAroundTarget()